src/__tests__/ceramic_cas.ts

```diff
@@ -0,0 +1,60 @@
+import {buildServicesFromConfig, Services} from "../index";
+import {AnchorStatus, CeramicApi} from "@ceramicnetwork/common"
+
+const registerChangeListener = function (doc: any): Promise<void> {
+    return new Promise(resolve => {
+        doc.on('change', () => {
+            resolve()
+        })
+    })
+}
+
+describe('Ceramic<->CAS integration', () => {
+    jest.setTimeout(1000 * 60 * 5) // 5 minutes
+    let ceramic: CeramicApi
+
+    beforeAll(async () => {
+        const services: Services = await buildServicesFromConfig();
+        ({ceramic} = services);
+    });
+
+    afterAll(async () => {
+        await ceramic.close()
+    });
+
+    test("basic crud", async () => {
+        // Test document creation
+        console.log("Creating document")
+        const initialContent = { foo: 'bar' }
+        const doc = await ceramic.createDocument('tile', {content: initialContent})
+        expect(doc.content).toEqual(initialContent)
+
+        // Test document creation is anchored correctly
```

Comment:
Do we still have a `processing` state? Or any other states? We should include tests for those too.

Reply:
We do, but I don't think there's any reasonable way to check for it in an integration test.  The processing state is pretty short-lived, so any check that tries to look for that state is going to be inherently racy

Reply:
How do we know the processing state is short lived?

The way this is currently written seems fragile--what if the state after pending is processing and not success? Does that break our expectations?

Also it seems we should test the failure state.

Reply:
Yeah, you're right about the test being fragile.  Whether we get back `processing` or `anchored` depends on the timing of when the http client polls.  Good catch.

I'll try to add some logic to handle that case, although it'll be hard to test that logic since I can't trigger that state on demand.  I guess I can run a modified version of CAS that hangs after putting requests into `processing` state to manually test for now.

As for testing the failure state, I agree, but that's another one that's hard to induce on demand.  There might be a way to do so by constructing an invalid update record, but we'd have to use some lower-level apis in js-ceramic to do so.  I'd like to leave that out of this PR for now and circle back to it later.  For now I'm mostly concerned with getting some end-to-end tests of everything actually working, before worrying too much about error handling.

Reply:
Okay @valmack, new version up that's robust to the PROCESSING state

Reply:
Yeah I think version of CAS that hangs will be a helpful test. And as for failures yeah I think it's ok to circle back if there is nothing obvious at the moment.

Reply:
> Yeah I think version of CAS that hangs will be a helpful test.

I did this locally when testing as I went, but there's no way to induce this behavior in the CAS programatically at the moment